### PR TITLE
sample(sample/22): Update graphql-prisma sample to prisma v3

### PR DESCRIPTION
**Pull request by TasinIshmam**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Other... Please describe: Update sample-22 to use the latest version of Prisma

## What is the current behavior?

The [graphql-prisma sample](https://github.com/nestjs/nest/tree/master/sample/22-graphql-prisma) uses Prisma version 1, which is in maintenance mode and no longer being actively developed ([announcement](https://github.com/prisma/prisma1/issues/4898)). 

Issue Number: N/A 

The [Contributing guidelines](https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md) mentioned there's no need to create a new issue before submitting a pull request for small changes, I hope this is appropriate.

## What is the new behavior?

The graphql-prisma sample has been updated to use the latest version of Prisma.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: sample/22-graphql-prisma/src/common/directives/upper-case.directive.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { defaultFieldResolver, GraphQLField } from 'graphql';
+
+export class UpperCaseDirective extends SchemaDirectiveVisitor {
+  visitFieldDefinition(field: GraphQLField<any, any>) {
+    const { resolve = defaultFieldResolver } = field;
+    field.resolve = async function (...args) {
+      const result = await resolve.apply(this, args);
+      if (typeof result === 'string') {
+        return result.toUpperCase();
+      }
```
Comment (kamilmysliwiec) [inline sample/22-graphql-prisma/src/common/directives/upper-case.directive.ts:11]: This file can be removed

File: sample/22-graphql-prisma/src/common/plugins/logging.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Plugin } from '@nestjs/graphql';
+import {
+  ApolloServerPlugin,
+  GraphQLRequestListener,
+} from 'apollo-server-plugin-base';
+
```
Comment (kamilmysliwiec) [inline sample/22-graphql-prisma/src/common/plugins/logging.plugin.ts:6]: This file can be removed

File: sample/22-graphql-prisma/src/posts/posts.module.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
+import { PostService } from './posts.service';
+import { PrismaModule } from '../prisma/prisma.module';
 
 @Module({
-  providers: [PostsResolver],
+  providers: [PostResolvers, PostService],
   imports: [PrismaModule],
 })
-export class PostsModule {}
+export class PostModule {}
```
Comment (kamilmysliwiec) [inline sample/22-graphql-prisma/src/posts/posts.module.ts:10]: nit: change to plural. Post -> Posts (similarly, `PostResolvers` -> `PostsResolver` etc)

File: sample/22-graphql-prisma/src/posts/posts.service.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  async post(id: string): Promise<Post | null> {
+    return this.prisma.post.findUnique({
+      where: {
+        id,
+      },
+    });
+  }
+
+  async posts(): Promise<Post[]> {
```
Comment (kamilmysliwiec) [inline sample/22-graphql-prisma/src/posts/posts.service.ts:22]:
```suggestion
  async findAll(): Promise<Post[]> {
```

File: sample/22-graphql-prisma/src/posts/posts.service.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import { PubSub } from 'graphql-subscriptions';
+import { Subscription } from '@nestjs/graphql';
+
+const pubSub = new PubSub();
+
+@Injectable()
+export class PostService {
+  constructor(private prisma: PrismaService) {}
+
+  async post(id: string): Promise<Post | null> {
```
Comment (kamilmysliwiec) [inline sample/22-graphql-prisma/src/posts/posts.service.ts:14]:
```suggestion
  async findOne(id: string): Promise<Post | null> {
```

File: sample/22-graphql-prisma/src/posts/posts.service.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        id,
+      },
+    });
+  }
+
+  async posts(): Promise<Post[]> {
+    return this.prisma.post.findMany({});
+  }
+
+  async createPost(input: NewPost): Promise<Post> {
```
Comment (kamilmysliwiec) [inline sample/22-graphql-prisma/src/posts/posts.service.ts:26]:
```suggestion
  async create(input: NewPost): Promise<Post> {
```

File: sample/22-graphql-prisma/src/posts/posts.service.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    const createdPost = await this.prisma.post.create({
+      data: input,
+    });
+    pubSub.publish('postCreated', { postCreated: createdPost });
+    return createdPost;
+  }
+
+  async updatePost(params: UpdatePost): Promise<Post> {
```
Comment (kamilmysliwiec) [inline sample/22-graphql-prisma/src/posts/posts.service.ts:34]:
```suggestion
  async update(params: UpdatePost): Promise<Post> {
```

File: sample/22-graphql-prisma/src/posts/posts.service.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  async deletePost(id: string): Promise<Post> {
+    return this.prisma.post.delete({
+      where: {
+        id,
+      },
+    });
+  }
+
+  @Subscription('postCreated')
+  catCreated() {
```
Comment (kamilmysliwiec) [inline sample/22-graphql-prisma/src/posts/posts.service.ts:56]:
```suggestion
  @Subscription()
  postCreated() {
```

File: sample/22-graphql-prisma/src/common/directives/upper-case.directive.ts (hunk 1 above)
Comment (TasinIshmam) [reply sample/22-graphql-prisma/src/common/directives/upper-case.directive.ts]: I removed the `common` directory because it was entirely unused.

**Comment by TasinIshmam** [comment]
@kamilmysliwiec thanks for the review. All suggested changes have been addressed.

**Comment by TasinIshmam** [comment]
Hey @kamilmysliwiec, could you kindly provide updates regarding this PR? 

If you have any other feedback or changes to suggest before merging, I'd be happy to work on those. 

Thanks!

**Comment by kamilmysliwiec** [comment]
Changes from this PR were cherry-picked here https://github.com/nestjs/nest/pull/9896. Thank yoU!

**Pull request closed without merging**